game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaPowerStrengthAndRolls.java

```diff
@@ -267,14 +268,15 @@ private void addUnits(final Collection<Unit> units) {
         .flatMap(
             unitPowerStrengthAndRolls -> {
               final int strength = unitPowerStrengthAndRolls.getStrength();
-              final int diceValue = diceQueue.removeFirst();
               return IntStream.range(0, unitPowerStrengthAndRolls.getRolls())
                   .mapToObj(
-                      rollNumber ->
-                          new Die(
-                              diceValue,
-                              strength,
-                              diceValue < strength ? Die.DieType.HIT : Die.DieType.MISS));
+                      rollNumber -> {
+                        final int diceValue = diceQueue.removeFirst();
```

Comment:
This fixes a bug I introduced when I refactored this code.  I now have better tests around this.  This bug is in the pre-release so I didn't add any release notes for it.